### PR TITLE
Fixed #4548 -- Added a username hint for password managers to admin's change_password form.

### PR DESCRIPTION
**Pull request by sliverc**
When more than one password is stored in the browser of a single site, a confirm password dialog pops up when changing password as browser doesn't know which user actually changed password.

This can be avoided by simply adding a username input field which is not displayed.

See also http://stackoverflow.com/a/2700066/535679

Successfully tested with Chromium and Firefox.

### CODE REVIEW

File: django/contrib/admin/templates/admin/auth/user/change_password.html [hunk 1]
```diff
@@ -20,6 +20,7 @@
 {% endif %}
 {% block content %}<div id="content-main">
 <form action="{{ form_url }}" method="post" id="{{ opts.model_name }}_form">{% csrf_token %}{% block form_top %}{% endblock %}
+<input type="text" name="username" value="{{ original.username }}" style="display: none" />
```
Comment (charettes) [inline django/contrib/admin/templates/admin/auth/user/change_password.html]: Does input `type="hidden"` works?

Comment (sliverc) [inline django/contrib/admin/templates/admin/auth/user/change_password.html]: It unfortunately doesn't. I tested it.

**Comment by charettes** [comment]
I wonder if it would be feasible to write a Selenium test for this.

File: django/contrib/admin/templates/admin/auth/user/change_password.html (hunk 1 above)
Comment (timgraham) [inline django/contrib/admin/templates/admin/auth/user/change_password.html]: also: original.get_username

File: tests/admin_views/tests.py [hunk 2]
```diff
@@ -818,6 +818,17 @@ def test_hide_change_password(self):
             msg_prefix='The "change password" link should not be displayed if a user does not have a usable password.'
         )
 
+    def test_template_change_password(self):
+        """
+        Ensure that the auth/user/change_password.html template has
```
Comment (timgraham) [inline tests/admin_views/tests.py]: Just state the expected behavior (no "Ensure that" prefix needed) and include the explanation about password managers. In that case, the ticket doesn't offer any additional info, so we can remove the reference to it.

**Comment by sliverc** [comment]
Code adjusted concerning the different comments from @timgraham 

Concerning selenium test:
To test the correct behavior what such a test needs to do is to check whether the confirm password dialog doesn't appear but instead the save password popup is shown with the correct username.
This is a bit tricky though as it is browser specific and this dialog is not part of the DOM. I do not really see how this can be accomplished with selenium without any additional browser plugins unless I have overlooked something?

**Comment by timgraham** [comment]
buildbot, add to whitelist.

**Comment by timgraham** [comment]
I agree that a selenium test would probably have to be browser specific and so probably isn't worth it.
merged in 5d8375fe662f636aff338a70ffa49644f9f53b39, thanks!

**Pull request closed without merging**